### ISSUE TITLE
Recursive references and uiSchema

### ISSUE THREAD

**Issue by mikebarkmin** [mozilla-services/react-jsonschema-form#670]
How do you handle recursive references with a custom uiSchema? This code does not work:

```javascript
const schema = {
  definitions: {
    block: {
      type: 'object',
      properties: {
        head: {
          type: 'string'
        },
        content: {
          type: 'string'
        },
        blocks: {
          type: 'array',
          items: {
            $ref: '#/definitions/block'
          }
        }
      }
    }
  },
  type: 'object',
  properties: {
    blocks: {
      type: 'array',
      items: {
        $ref: '#/definitions/block'
      }
    }
  }
};

const uiSchema = {
  definitions: {
    block: {
      content: {
        'ui:widget': 'textarea'
      },
      blocks: {
        items: {
          $ref: '#/definitions/block'
        }
      }
    }
  },
  blocks: {
    items: {
      $ref: '#/definitions/block'
    }
  }
};
```

**Comment by n1k0** [comment]
This is being addressed in https://github.com/mozilla-services/react-jsonschema-form/pull/557

**Comment by phemmer** [comment]
Was this ever addressed? I see it was closed due to #557, but #557 was also closed, and I see no other issues on the subject.
